examples/miscellaneous/plot_outlier_detection_bench.py

```diff
@@ -0,0 +1,186 @@
+"""
+==========================================
+Evaluation of outlier detection estimators
+==========================================
+
+This example benchmarks outlier detection algorithms, Local Outlier
+Factor (LOF) and Isolation Forest (IForest), using ROC curves on
+classical anomaly detection datasets. The algorithm performance
+is assessed in an outlier detection context:
+
+1. The algorithms are trained on the whole dataset which is assumed to
+contain outliers.
+
+2. The ROC curve is computed on the same dataset using the knowledge
+of the labels.
+
+There is no particular reason to choose algorithms LOF and IForest. The goal
+is to show that different algorithm performs well on different datasets.
+"""
+
+# Author: Pharuj Rajborirug <pharuj.ra@kmitl.ac.th>
+# License: BSD 3 clause
+
+import numpy as np
+
+print(__doc__)
+
+rng = np.random.RandomState(42)
+
+# %%
+# Generate datasets
+# -----------------
+#
+# The example uses real world datasets available in `scikit-learn.datasets`.
+datasets = [
+    "http",
+    "smtp",
+    "SA",
+    "SF",
+    "forestcover",
+    "glass",
+    "wdbc",
+    "cardiotocography",
+]
```

Comment:
I think it would be clearer if the data loading was actually in this section.

We could define a function preprocess_dataset that takes the dataset name. It would load the dataset and preprocess it and then returns X, y.

Reply:
Currently, I loaded, preprocessed, and stored datasets lists Xs = [X1, X2,..] and Ys. It requires more memory but will finish all data preprocessing here. In the next code block, we will just call the data to compute ROV. Do you prefer this method?

Reply:
> It requires more memory

That's why I suggested to define a function to load and preprocess one dataset at a time. Then in the next cells we'll call the function for each dataset

Reply:
Done it! I created 2 functions. one for the data processing and the other for the model prediction. The example will only call each function when needed.